src/renderers/server-render.jsx

```diff
@@ -187,24 +191,13 @@ const makeRenderer = (
 	})
 	.first(state => state.preRenderChecklist.every(isReady => isReady));  // take the first ready state
 
-	request.log(['app', 'info'], `Finding route for path: '${appLocation}'`);
-	return Rx.Observable.bindNodeCallback(match)({ location: appLocation, routes })
-		.do(([redirectLocation, renderProps]) => {
-			if (!redirectLocation && !renderProps) {
-				throw Boom.notFound();
-			}
-		})
-		.do(() => {
-			request.log(['app', 'info'], 'Found app route, dispatching RENDER');
-		})
-		.do(([redirectLocation, renderProps]) =>
-			store.dispatch({
-				type: '@@server/RENDER',
-				payload: renderProps.location
-			})
-		)
-		.flatMap(args => storeIsReady$.map(() => args))  // `sample` appears not to work - this is equivalent
-		.map(getRouterRenderer(store, baseUrl, clientFilename, assetPublicPath));
+	request.log(['app', 'info'], `Finding route for path: '${url}'`);
+	store.dispatch({
+		type: '@@server/RENDER',
+		payload: request.url,
+	});
+	return storeIsReady$
+		.map(() => getRouterRenderer(routes, store, url, baseUrl, clientFilename, assetPublicPath));
```

Comment:
yeah, as soon as I started writing `<AppContainer />` for the starter kit, I realized that it would essentially be identical to `RouteLayout` for every application, so we can just keep the `getRenderer` API stable and use a platform-hosted `RouteLayout`.

In the future, I suspect we might want allow apps to pass in a 'custom' `<App />` that implements its own routing, but since the routing config is so closely tied to API requests, I don't think we're ready to support that functionality.